File: java/src/main/java/mx/iteso/adapter/GoogleDoc.java
package mx.iteso.adapter;

import java.awt.Font;
/**Google doc class. */
public class GoogleDoc implements IGoogleDoc {
    /**Permisos. 1 = Solo ver, 2 = Puede comentar, 3 = Puede editar */
    private int sharingPermissions;
    /**Font. */
    private Font font;
    /**Style using format class. */
    private Format style;
    /**Background image. */
    private BackgroundImage bImage;
    /**font size. */
    private static final int FONT_SIZE = 20;
    /**image width. */
    private static final int IMAGE_WIDTH = 1024;
    /**image height. */
    private static final int IMAGE_HEIGHT = 720;
    /**Constructor. */
    public GoogleDoc() {
        this.font = new Font("TimesRoman", Font.PLAIN, FONT_SIZE);
        this.bImage = new BackgroundImage("bgImage.jpg",
        IMAGE_WIDTH, IMAGE_HEIGHT);
        this.style = new Format("Arial", 12);
    }
    /**
     * @return font.
     */
    public Font getFont() {
        return this.font;
    }
    /**
     * @return style.
     */
    public Format getStyle() {
        return this.style;
    }
    /**
     * @return background image.
     */
    public BackgroundImage getBackground() {
        return this.bImage;
    }
    /**
     * @return sharing permissions.
     */
    public int getSharingPermissions() {
        return this.sharingPermissions;
    }
    /**
     * Permissions.
     * @param sharingPermissions2 permissions.
     */
    public void setSharingPermissions(final int sharingPermissions2) {
        this.sharingPermissions = sharingPermissions2;
    }
}

File: python/src/main/python/GoogleDoc.py
from IGoogleDoc import *
from Format import *
from BackgroundImage import *
from Font import *


class GoogleDoc(IGoogleDoc):

    def __init__(self):
        self.font = Font("Timesroman", 12)
        self.style = Font("Arial", 12)
        self.backgroundImage = BackgroundImage("image", 1024.0, 720.0)
        self.sharingPermissions = 1

    def getFont(self):
        return self.font

    def getStyle(self):
        return self.style

    def getBackground(self):
        return self.backgroundImage

    def getSharingPermissions(self):
        return self.sharingPermissions

    def setSharingPermissions(self, sharingPermissions2):
        self.sharingPermissions = sharingPermissions2
